src/Contrib/Jaeger/SpanConverter.php

```
@@ -0,0 +1,190 @@
+            'vType' => TagType::STRING,
+            'vStr' => $value,
+        ]);
+
+        //TODO - determine if any of this additional handling is still needed, or if it's all dead-code (i.e. $value must be a string at this point always)
+
+        // if (is_bool($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::BOOL,
+        //         'vBool' => $value,
+        //     ]);
+        // } elseif (is_string($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::STRING,
+        //         'vStr' => $value,
+        //     ]);
+        // } elseif (null === $value) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::STRING,
+        //         'vStr' => '',
+        //     ]);
+        // } elseif (is_integer($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::LONG,
+        //         'vLong' => $value,
+        //     ]);
+        // } elseif (is_numeric($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::DOUBLE,
+        //         'vDouble' => $value,
+        //     ]);
+        // }
+
+        // error_log('Cannot build tag for ' . $key . ' of type ' . gettype($value));
+
+        // throw new \Exception('unsupported tag type');
+    }
+
+    private static function convertOtlpToJaegerIds(SpanDataInterface $span): array
+    {
+        if (PHP_INT_SIZE < 8) {
+            $humanReadableIntSize = PHP_INT_SIZE*8;
+
+            throw new RuntimeException("Integrating with Jaeger requires usage of 64 bit integers, but your current platform is $humanReadableIntSize bit. See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#ids for more information.");
```

I think this belongs in a constructor (which you'll need to add) - better to fail fast if we can.

---

Sounds good to me, moved it into one